supporting-blog-content/hybrid-search-for-an-e-commerce-product-catalogue/product-store-search/api/api.py

```diff
@@ -0,0 +1,199 @@
+import yaml
+from elasticsearch import Elasticsearch
+from flask import Flask, jsonify, request
+from flask_cors import CORS
+from sentence_transformers import SentenceTransformer
+
+app = Flask(__name__)
+CORS(app)
+
+promote_products_free_gluten = ["1043", "1042", "1039"]
+
+
+def get_client_es():
+    with open('../config.yml', 'r') as file:
+        config = yaml.safe_load(file)
+    return Elasticsearch(
+        cloud_id=config['cloud_id'],
+        api_key=config['api_key']
+    )
+
+
+def get_text_vector(sentences):
+    model = SentenceTransformer('sentence-transformers/all-MiniLM-L6-v2')
+    embeddings = model.encode(sentences)
+    return embeddings
+
+
+def build_query(term=None, categories=None, product_types=None, brands=None):
+    must_query = [{"match_all": {}}] if not term else [{
+        "multi_match": {
+            "query": term,
+            "fields": ["name", "category", "description"]
+        }
+    }]
+
+    filters = []
+    if categories:
+        filters.append({"terms": {"category": categories}})
+    if product_types:
+        filters.append({"terms": {"product_type": product_types}})
+    if brands:
+        filters.append({"terms": {"brand.keyword": brands}})
+
+    return {
+        "_source": ["id", "brand", "name", "price", "currency", "image_link", "category", "tag_list"],
+        "query": {
+            "bool": {
+                "must": must_query,
+                "filter": filters
+            }
+        }
+    }
+
+
+def build_hybrid_query(term=None, categories=None, product_types=None, brands=None, hybrid=False):
+    # Standard query
+    organic_query = build_query(term, categories, product_types, brands)
+
+    if hybrid is True and term:
+
+        vector = get_text_vector([term])[0]
+
+        # Hybrid query with RRF (Reciprocal Rank Fusion)
+        query = {
+            "retriever": {
+                "rrf": {
```

Comment:
@andreluiz1987 thanks for your [post](https://www.elastic.co/search-labs/blog/hybrid-search-ecommerce) and contribution! it was great!
What are the consideration for chosing `rrf` over a simple `should`?
Do you get the same score range with `rrf`? (`knn` return `_score` between 0-1.0 and `organic_query` doesn't have an upper `_score` bound)

I asked in the [forum](https://discuss.elastic.co/t/normalizing-knn-and-multi-match-clauses/372307) but I wondered if you have the same issue here.